packages/toc/src/toc_item.tsx

```diff
@@ -2,7 +2,92 @@
 // Distributed under the terms of the Modified BSD License.
 
 import * as React from 'react';
-import { IHeading } from './utils/headings';
+import { IHeading, INotebookHeading } from './utils/headings';
+import { CodeCell } from '@jupyterlab/cells';
+import { NotebookPanel } from '@jupyterlab/notebook';
+import { Signal } from '@lumino/signaling';
+import { TableOfContents } from './toc';
+
+/**
+ * Tests whether a heading is a notebook heading.
+ *
+ * @param heading - heading to test
+ * @returns boolean indicating whether a heading is a notebook heading
+ */
+const isNotebookHeading = (heading: any): heading is INotebookHeading => {
+  return heading.type !== undefined && heading.cellRef !== undefined;
+};
+
+/**
+ * Checks whether a heading has runnable code cells.
+ * Runs runnable code cells if given runCode.
+ *
+ * @private
+ * @param headings - list of headings
+ * @param heading - heading
+ * @param runCode - optional empty list of "code" headings
+ * @returns boolean indicating whether a heading has runnable code cells
+ */
+export const NestedCodeCells = (
```

Comment:
If this is a function I think it could be better to put it in the `utils` directory and name it something like `runNestedCodeCells`. I don't think it needs to return a boolean either

Reply:
@marthacryan thanks so so much for the help! I am moving `runNestedCodeCells` to `utils` and ran into `Can't resolve '@jupyterlab/toc/src/utils/headings'` when I try to import from `toc/src/utils` to `toc-extension/src/index.ts `. Do you know why that might be?

Reply:
Oh you'll probably want to add an export line to the `index.ts` of the `toc` package. Then you'll just import `runNestedCodeCells` from `@jupyterlab/toc` without the extra paths

Reply:
I see, make sense :)